src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -455,6 +455,10 @@ def query(self, filters=None, options=None):
         `query-options.extra.snapshots` is a boolean which when set will retrieve snapshots for the dataset in question
         by adding a snapshots key to the dataset data.
 
+        `query-options.extra.retrieve_children` is a boolean which when set which is the default will retrieve child
+        datasets of a dataset. Motivation for providing a knob to configure this is performance concerns where getting
+        children sometimes is not required and in that case this user can be explicitly unset.
+
```

Comment:
Just so I'm understanding correctly, this will not retrieve grandchildren datasets properties? `/mnt/tank/parent/child/grandchild`?

Reply:
It will not traverse children at all which means yes it won't retrieve properties for them as well as the traversal would never happen and we remove `children` key altogether to not give the impression that this dataset does not has children when it might have.

Reply:
So it will only retrieve top-level datasets? `/mnt/tank/a` and `/mnt/tank/b` and `/mnt/tank/b` etc?

Reply:
If it only retrieves the top-level datasets then may I suggest a better doc-string?

`query-options.extra.retrieve_children is a boolean. Set to True by default. When set to True, will retrieve all child datasets which can cause a performance penalty. When set to False, will explicitly ignore all child datasets and only return the top-level datasets.`

Reply:
Sure, so the bit about `top-level datasets` is not accurate as it retrieves the datasets passed and that can be `pool/someds/anotherchild` which is not a top level dataset. Can you please advise on that ? Thank you

Reply:
Sure,

`query-options.extra.retrieve_children is a boolean. Set to True by default. When set to True, will retrieve all children datasets which can cause a performance penalty. When set to False, will not retrieve children datasets which does not incur the performance penalty.`